File: src/main/java/com/server/ordering/domain/member/Customer.java
package com.server.ordering.domain.member;

import com.server.ordering.domain.*;
import lombok.*;

import javax.persistence.*;

import java.util.ArrayList;
import java.util.List;

import static javax.persistence.CascadeType.*;
import static javax.persistence.FetchType.*;
import static javax.persistence.GenerationType.*;
import static lombok.AccessLevel.*;

@Entity
@Getter
@NoArgsConstructor(access = PROTECTED)
public class Customer extends MemberBase {

    @Id @GeneratedValue(strategy = IDENTITY)
    @Column(name = "customer_id")
    private Long id;

    private String nickname;

    private Long basketKey;
    private String firebaseToken;

    @OneToMany(mappedBy = "customer", cascade = REMOVE, orphanRemoval = true)
    private final List<Basket> baskets = new ArrayList<>();

    @OneToMany(mappedBy = "customer", cascade = REMOVE, orphanRemoval = true)
    private final List<Order> orders = new ArrayList<>();

    @OneToMany(mappedBy = "customer", cascade = REMOVE, orphanRemoval = true)
    private final List<MyCoupon> coupons = new ArrayList<>();

    @OneToOne(mappedBy = "customer", cascade = REMOVE, orphanRemoval = true, fetch = LAZY)
    private Waiting waiting;

    public Customer(String nickname, String signInId, String password, PhoneNumber phoneNumber) {
        super(signInId, password, phoneNumber);
        this.nickname = nickname;
    }

    public void changeBasketKey(Long restaurantId) {
        this.basketKey = restaurantId;
    }

    public Boolean isAbleToWaiting() {
        return this.waiting == null;
    }

    public void putFirebaseToken(String firebaseToken) {
        this.firebaseToken = firebaseToken;
    }

    public void clearFirebaseToken() {
        this.firebaseToken = null;
    }
}

File: src/main/kotlin/com/byeoru/ordering_server/domain/member/Customer.kt
package com.byeoru.ordering_server.domain.member

import com.byeoru.ordering_server.domain.*
import java.util.ArrayList
import javax.persistence.*

@Entity
class Customer(nickname: String,
               signInId: String,
               password: String,
               phoneNumber: PhoneNumber) : MemberBase(signInId, password, phoneNumber) {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "customer_id")
    var id: Long? = null
         protected set

     var nickname: String = nickname
         protected set

     var basketKey: Long? = null
         protected set

    @OneToMany(mappedBy = "customer", cascade = [CascadeType.REMOVE])
     var bookmarks: MutableList<Bookmark> = ArrayList<Bookmark>()
         protected set

    @OneToMany(mappedBy = "customer", cascade = [CascadeType.REMOVE], orphanRemoval = true)
     var baskets: MutableList<Basket> = ArrayList<Basket>()
         protected set

    @OneToMany(mappedBy = "customer")
     var orders: MutableList<Order> = ArrayList<Order>()
         protected set

    @OneToMany(mappedBy = "customer", cascade = [CascadeType.REMOVE], orphanRemoval = true)
     var coupons: MutableList<MyCoupon> = ArrayList<MyCoupon>()
         protected set

    @OneToOne(mappedBy = "customer", cascade = [CascadeType.REMOVE], orphanRemoval = true)
     var waiting: Waiting? = null
         protected set

    fun changeBasketKey(restaurantId: Long?) {
        basketKey = restaurantId
    }

    fun isAbleToWaiting() = waiting == null
}